x-pack/packages/ml/is_populated_object/README.md

```diff
@@ -0,0 +1,3 @@
+# @kbn/ml-is-populated-object
+
+ A type guard to check record like object structures.
```

Comment:
Wonder if we need to expand the description of this readme more... Noticed because of how the imports/packages are structured now, clicking on `isPopulatedObject` in Webstorm directs to the types without any hint for what the function does instead of [the actual file where the description is](https://github.com/elastic/kibana/blob/d657ce46f2d506ce1f4691ce051058817796ea4d/x-pack/packages/ml/is_populated_object/src/is_populated_object.ts). Does it work correctly in VSC? Possibly just an issue on my set up.

Reply:
Found out that the comments for the function was missing a `*` for the starting `/**` so it wasn't picked up correctly. I pushed a fix so it should now be part of the `d.ts` file and show up in IDEs:

<img width="50%" alt="image" src="https://user-images.githubusercontent.com/230104/175022215-ddaf10ef-4aba-4c2f-83e1-aa538d3adfcb.png">

I also added a task to the packages that auto-generates markdown docs in the `README.md` files of the packages. It picks up JSDoc comments and looks like this for `isPopulatedObject()`:

---

# @kbn/ml-is-populated-object

<!-- INSERT GENERATED DOCS START -->

### `isPopulatedObject` (function)

A type guard to check record like object structures.

Examples:

- `isPopulatedObject({...})`
  Limits type to Record<string, unknown>

- `isPopulatedObject({...}, ['attribute'])`
  Limits type to Record<'attribute', unknown>

- `isPopulatedObject<keyof MyInterface>({...})`
  Limits type to a record with keys of the given interface.
  Note that you might want to add keys from the interface to the
  array of requiredAttributes to satisfy runtime requirements.
  Otherwise you'd just satisfy TS requirements but might still
  run into runtime issues.

<!-- INSERT GENERATED DOCS END -->

Reply:
Fantastic!!! Amazing work, this definitely improves the dev experience a lot! Also thanks for catching the missing *.